sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -1224,188 +1377,1047 @@ public async Task UpdateAzureDataLakeStorageGen2DataFeedWithMinimumSetupAndNewIn
         {
             MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
 
             var originalConnectionString = "connectionStr";
-            var originalTable = "table";
             var originalQuery = "query";
 
             var dataFeedName = Recording.GenerateAlphaNumericId("dataFeed");
-            var dataSource = new AzureTableDataFeedSource(originalConnectionString, originalTable, originalQuery);
-            var description = "This data feed was created to test the .NET client.";
+            var dataSource = new AzureDataExplorerDataFeedSource(originalConnectionString, originalQuery);
             DataFeed dataFeedToCreate = GetDataFeedWithMinimumSetup(dataFeedName, dataSource);
 
             await using var disposableDataFeed = await DisposableDataFeed.CreateDataFeedAsync(adminClient, dataFeedToCreate);
 
             DataFeed dataFeedToUpdate = GetDataFeedWithMinimumSetup(dataFeedName, dataSource);
 
-            dataFeedToUpdate.Description = description;
+            SetOptionalMembers(dataFeedToUpdate);
 
             await adminClient.UpdateDataFeedAsync(disposableDataFeed.Id, dataFeedToUpdate);
 
             DataFeed updatedDataFeed = await adminClient.GetDataFeedAsync(disposableDataFeed.Id);
 
-            ValidateDataFeedWithMinimumSetup(updatedDataFeed, disposableDataFeed.Id, dataFeedName, description);
+            ValidateUpdatedDataFeedWithOptionalMembersSet(updatedDataFeed, disposableDataFeed.Id, dataFeedName);
 
-            Assert.That(updatedDataFeed.SourceType, Is.EqualTo(DataFeedSourceType.AzureTable));
+            Assert.That(updatedDataFeed.SourceType, Is.EqualTo(DataFeedSourceType.AzureDataExplorer));
+
+            var updatedDataSource = updatedDataFeed.DataSource as AzureDataExplorerDataFeedSource;
+
+            var expectedConnectStr = TestEnvironment.Mode == RecordedTestMode.Playback ? "Sanitized" : originalConnectionString;
+
+            Assert.That(updatedDataSource, Is.Not.Null);
+            Assert.That(updatedDataSource.ConnectionString, Is.EqualTo(expectedConnectStr));
+            Assert.That(updatedDataSource.Query, Is.EqualTo(originalQuery));
+        }
+
+        [RecordedTest]
+        public async Task UpdateAzureDataLakeStorageGen2DataFeedWithMinimumSetupAndGetInstance()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            var originalAccount = "account";
+            var originalKey = "key";
+            var originalFileSystem = "fileSystem";
+            var originalDirectory = "dir";
+            var originalFile = "file";
+
+            var dataFeedName = Recording.GenerateAlphaNumericId("dataFeed");
+            var dataSource = new AzureDataLakeStorageGen2DataFeedSource(originalAccount, originalKey, originalFileSystem, originalDirectory, originalFile);
+            var description = "This data feed was created to test the .NET client.";
```

Comment:
This could be const and defined for the whole class as well.

Reply:
I'm keeping the "DataFeed-level" properties local so the test class don't get too crowded. I agree about making it const.